### PR TITLE
Misc pipeline improvements

### PR DESCRIPTION
**Pull request by elfeck**
This PR addresses #172 (still contains valgrind-errors) and parts of #123. I'm running all boost tests on my machine right now to see all is well.

After this merges #172 and #123 can probably be closed. I will discuss with @krsgpp if there should be any new issues with left-over stuff.

### CODE REVIEW

**Comment by elfeck** [comment]
After this PR is merged, the wiki should update the configuration section (I will make sure all wiki upstream changes are merged before that). [This file](https://github.com/elfeck/SGpp-wiki/blob/new_update/Datadriven-datamining-pipeline-configuration.md) contains the new version.

**Comment by krsgpp** [comment]
Travis fails with
> datadriven/build/jsgpp/datadriven.i:25: Error: Unable to find 'datadriven/src/sgpp/datadriven/configuration/BatchConfiguration.hpp'

**Comment by elfeck** [comment]
This BatchConfiguration issue is now fixed. Now failing a python test because time exceeded.

**Comment by valentjn** [comment]
@elfeck FYI, empty commits aren't necessary to restart CI checks. You may ask your advisor to click the restart buttons in Travis and/or Jenkins, or (probably as ugly as empty commits) you may rebase your last commit (probably with `rebase --ignore-date HEAD~`) and force-push it.

**Comment by elfeck** [comment]
@valentjn Thanks for the info. Finally the test passed, probably due to the change in master which I just merged.

**Comment by krsgpp** [comment]
[Jenkins Style Job](https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-style-testing/165/warnings14Result/) is not happy.

Errors in the openmpi job are
> unknown location(0): fatal error in "testOnOffParallelChol": std::exception: Not enough processes available to form BLACS process grid!

and similar. I think, those are not related to this PR, but maybe to hardware limits? @leiterrl

**Comment by elfeck** [comment]
I will fix the style issues

**Comment by krsgpp** [comment]
bump

**Comment by elfeck** [comment]
@krsgpp I can't find the meta-errors in the previous Jenkins job. Could you start the "Expected one" for me?

**Comment by krsgpp** [comment]
[Style Warnings in Jenkins](https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-style-testing/183/warnings14Result/)

**Comment by elfeck** [comment]
I merged upstream master and (hopefully) fixed the style issues

**Comment by krsgpp** [comment]
Looks good, thank you! The errors in the OpenMPI Job are not from you I think and the travis errors are the time limits also not caused by you.

**Comment by leiterrl** [comment]
> [Jenkins Style Job](https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-style-testing/165/warnings14Result/) is not happy.
> 
> Errors in the openmpi job are
> 
> > unknown location(0): fatal error in "testOnOffParallelChol": std::exception: Not enough processes available to form BLACS process grid!
> 
> and similar. I think, those are not related to this PR, but maybe to hardware limits? @leiterrl

Sorry for the late reply but this exception was added in 8e1a74192d7b5cfde8454000306cc981d3f0081c which got merged in #112 . I don't know why the test just started failing now but BlacsProcessGrid::rows and BlacsProcessGrid::columns is potentially set to 32 through the gmm_on_off_parallel_chol.json config. Maybe someone with more insight on what is actually going on here can chime in on that? Is @janschopohl perhaps still working with SGpp?

**Comment by janschopohl** [comment]
If the columns and rows for the BlacsProcessGrid are explicitly set in the config, an exception will be thrown if less than columns*rows processes are available.
I would suggest deleting processRows and processColumns from the json, then a square process grid with as many processes as possible is created (usually 1*1 in jenkins). This should be fine for testing on jenkins.
See the wiki for the configuration options: https://github.com/SGpp/SGpp/wiki/Datadriven-datamining-pipeline-configuration#parallelconfig

**Comment by elfeck** [comment]
Even though this is issue is not introduced with this PR (as far as I can see) I will fix now in the config @leiterrl mentioned above

**Comment by elfeck** [comment]
I'm confused. I did push the merge and resolve these conflicts. What happend?

**Comment by elfeck** [comment]
@krsgpp You pushed some changes into my branch, probably that's why. Should I do anything more in this PR or are you taking care of it?

**Comment by elfeck** [comment]
Okay seems like the issue was on my end after all. For some reason the commit and push or the merge did not go through.

File: datadriven/src/sgpp/datadriven/datamining/modules/fitting/FitterConfiguration.cpp [hunk 1]
```diff
@@ -147,6 +147,8 @@ void FitterConfiguration::setupDefaults() {
   // to have all of the default value config in this file.
   densityEstimationConfig.type_ = sgpp::datadriven::DensityEstimationType::Decomposition;
   densityEstimationConfig.decomposition_ = sgpp::datadriven::MatrixDecompositionType::OrthoAdapt;
```
[the remark this answers is not in the document]
Comment (krsgpp) [reply datadriven/src/sgpp/datadriven/datamining/modules/fitting/FitterConfiguration.cpp]: ok!

**Pull request merged**